lib/widgets/color.dart

```diff
@@ -38,4 +38,21 @@ extension ColorExtension on Color {
            ((g * 255.0).round() & 0xff) << 8 |
            ((b * 255.0).round() & 0xff) << 0;
   }
+
+  /// Makes a copy of this color with [a] multiplied by `factor`.
+  ///
+  /// `factor` must not be less than 0 or greater than 1.
+  ///
+  /// To fade a color variable from [DesignVariables], [ContentTheme], etc.,
+  /// use this instead of calling [withValues] with `factor` passed as `alpha`,
+  /// which simply replaces the color's [a] instead of multiplying by it.
+  /// Using [withValues] gives the same result for an opaque color,
+  /// but a wrong result for a semi-transparent color,
+  /// and we want our color variables to be free to change
+  /// without breaking things.
+  Color withFadedAlpha(double factor) {
```

Comment:
```
color [nfc]: Add ColorExtension.withFadedAlpha

We use this instead at most places where we preivously have
`.withValues(alpha:` on color variables, except the one in
`lib/widgets/emoji_reaction.dart` that is semi-transparent,
to keep this change NFC.

This also skips the ones in `lib/widgets/channel_colors.dart`,
and the one in `lib/widgets/lightbox.dart`
(`Colors.grey.shade900.withValues(alpha: 0.87)`), as determining if
those can be swapped out with `withFadedAlpha` seems out of scope as of
now.

Signed-off-by: Zixuan James Li <zixuan@zulip.com>
```

Commit message nits:

In the first paragraph, I'm not sure what "instead" refers to. How about: "And use it everywhere we were using `.withValues(alpha:` to fade a color variable, except for one place in lib/widgets/emoji_reaction.dart where the variable is semitransparent to begin with. We'll do that as a non-NFC change in a later commit.

About the second paragraph:
- I think the `withValues` calls in lib/widgets/channel_colors.dart should stay. The linked design for the computation (a replit by Vlad) doesn't specify multiplication; it says to set the alpha channel to 0.3, which we're doing with `withValues`.
- I think the `withValues` in lightbox.dart can stay. It's setting an alpha channel on a constant from Flutter (`Colors.grey.shade900`) that realistically won't ever change to be semi-transparent. Even if it did, it's not clear that 0.87 would be the wrong value to end up with.
- Like with the many `withValues` calls in lib/widgets/theme.dart, I think these don't need to be specifically mentioned in the commit message. They're using `withValues(alpha:` as intended, to create a color with its alpha channel set to a value. They might have done that same thing a different way, like in the `Color` constructor. That's different from "fad[ing] a color variable from [DesignVariables] […]", which we often do e.g. to translate from the Figma an `opacity` attribute on an element that's colored with a variable.

Reply:
Thanks! Changed the message to
```
And use it everywhere we were using `.withValues(alpha:`
to fade a color variable, except for one place in
`lib/widgets/emoji_reaction.dart` where the variable is
semi-transparent in dark mode to begin with.
We'll do that as a non-NFC change in a later commit.
```
with a minor tweak mentioning that, in dark mode, the color is semi-transparent.